crates/driver/src/domain/competition/score.rs

```diff
@@ -54,19 +41,45 @@ impl num::Zero for Score {
     }
 }
 
-/// Represents the probability that a solution will be successfully settled.
-#[derive(Debug, Copy, Clone)]
-pub struct SuccessProbability(pub f64);
+/// Represents the observed quality of a solution. This is not an artifical
+/// value like score. This is a real value that solution provides and it's
+/// defined as surplus + fees.
+#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+pub struct Quality(pub eth::U256);
 
-impl From<f64> for SuccessProbability {
-    fn from(value: f64) -> Self {
+impl From<eth::U256> for Quality {
+    fn from(value: eth::U256) -> Self {
         Self(value)
     }
 }
 
+/// ObjectiveValue = Quality - GasCost
+impl std::ops::Sub<GasCost> for Quality {
+    type Output = ObjectiveValue;
+
+    fn sub(self, other: GasCost) -> Self::Output {
+        ObjectiveValue(self.0.saturating_sub(other.0 .0))
+    }
+}
```

Comment:
I don't really like to silently overload `ObjectiveValue(0)` to mean `ObjectiveValue(≤0)`.
It means that you always need to know if the objective value you are using has been "cleaned" by the function you are using to ignore zero or not.

What could be nice is that if we have an `ObjectiveValue` then it's also valid. We can say that this implementation of `Add` return an `Option<ObjectiveValue>` instead , which is `None` if `objective_value ≤ 0`.

Reply:
Ah, It slipped my mind completely that I can define Output as Option :)
Thanks, updated https://github.com/cowprotocol/services/pull/2022/commits/39858e03f8c9b0bcdec8ead269ae5f094584db78